lib/taskana-core/src/test/java/pro/taskana/impl/integration/WorkbasketServiceImplIntAutocommitTest.java

```diff
@@ -194,30 +190,30 @@ public void testUpdateWorkbasket() throws Exception {
 
         String id2 = IdGenerator.generateWithPrefix("TWB");
         Workbasket workbasket2 = createTestWorkbasket(id2, "key2", "novatec", "Hyperbasket", WorkbasketType.GROUP);
-        workbasket2.setDistributionTargets(new ArrayList<>());
-        workbasket2.getDistributionTargets().add(workbasket0.asSummary());
-        workbasket2.getDistributionTargets().add(workbasket1.asSummary());
         workbasket2 = workBasketService.createWorkbasket(workbasket2);
         createWorkbasketWithSecurity(workbasket2, "Elena", true, true, false, false);
+        List<String> distTargets = new ArrayList<>(Arrays.asList(workbasket0.getId(), workbasket1.getId()));
+        workBasketService.setDistributionTargets(workbasket2.getId(), distTargets);
 
         String id3 = IdGenerator.generateWithPrefix("TWB");
         Workbasket workbasket3 = createTestWorkbasket(id3, "key3", "novatec", "hm ... irgend ein basket",
             WorkbasketType.GROUP);
         workbasket3 = workBasketService.createWorkbasket(workbasket3);
         createWorkbasketWithSecurity(workbasket3, "Elena", true, true, false, false);
 
-        workbasket2.getDistributionTargets().clear();
-        workbasket2.getDistributionTargets().add(workbasket3.asSummary());
-        Thread.sleep(SLEEP_TIME);
-        workbasket2 = workBasketService.updateWorkbasket(workbasket2);
+        List<String> newDistTargets = new ArrayList<>(Arrays.asList(workbasket3.getId()));
+        workBasketService.setDistributionTargets(workbasket2.getId(), newDistTargets);
 
         Workbasket foundBasket = workBasketService.getWorkbasket(workbasket2.getId());
 
-        List<WorkbasketSummary> distributionTargets = foundBasket.getDistributionTargets();
+        List<WorkbasketSummary> distributionTargets = workBasketService.getDistributionTargets(foundBasket.getId());
+
         Assert.assertEquals(1, distributionTargets.size());
         Assert.assertEquals(id3, distributionTargets.get(0).getId());
-        Assert.assertNotEquals(workBasketService.getWorkbasket(id2).getCreated(),
-            workBasketService.getWorkbasket(id2).getModified());
+        // Question: should we update the modfied timestamp on the source workbasket if we change its
+        // distributiontargets?
+        // Assert.assertNotEquals(workBasketService.getWorkbasket(id2).getCreated(),
+        // workBasketService.getWorkbasket(id2).getModified());
```

Comment:
I guess we should change modified, because it´s a insert/update call to the DB. Can this be fixed?